src/test/java/com/ecse437/project/ProjectApplicationTests.java

```diff
@@ -1,13 +1,85 @@
 package com.ecse437.project;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 
+import com.ecse437.project.common.Common.Algorithm;
+import com.ecse437.project.controller.ProjectController;
+import com.ecse437.project.exceptions.IllegalAlgorithmException;
+import com.ecse437.project.services.Binary;
+import com.ecse437.project.services.Factorial;
+import com.ecse437.project.services.Hexa;
+import com.ecse437.project.services.Main;
+import com.ecse437.project.services.Prime;
+import com.ecse437.project.services.Squared;
+
 @SpringBootTest
 class ProjectApplicationTests {
 
+	@Autowired
+	private ProjectController controller;
+	
+	@Mock
+	private Binary binary;
+	
+	@Mock
+	private Factorial factorial;
+	
+	@Mock
+	private Hexa hexa;
+	
+	@Mock
+	private Prime prime;
+	
+	@Mock
+	private Squared squared;
+	
 	@Test
 	void contextLoads() {
+		assertThat(controller).isNotNull();
+	}
+	
+	@Test
+	void binaryCalculation() {
+		binary = new Binary();
+		assertThat(binary.compute(255).toString()).isEqualTo("11111111");
```

Comment:
Also worth testing the `print` function to test that the formatting is right

Reply:
Tested the print function for each function